packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -1989,14 +1997,20 @@ class _OverlayPortalState extends State<OverlayPortal> {
   @override
   Widget build(BuildContext context) {
     final int? zOrderIndex = _zOrderIndex;
-    if (zOrderIndex == null) {
-      return _OverlayPortal(overlayLocation: null, overlayChild: null, child: widget.child);
-    }
-    return _OverlayPortal(
-      overlayLocation: _getLocation(zOrderIndex, widget._targetRootOverlay),
-      overlayChild: _DeferredLayout(child: Builder(builder: widget.overlayChildBuilder)),
-      child: widget.child,
-    );
+    final _OverlayPortal portal =
+        zOrderIndex == null
+            ? _OverlayPortal(overlayLocation: null, overlayChild: null, child: widget.child)
+            : _OverlayPortal(
+              overlayLocation: _getLocation(zOrderIndex, widget._targetRootOverlay),
+              overlayChild: _DeferredLayout(child: Builder(builder: widget.overlayChildBuilder)),
+              child: widget.child,
+            );
+    // Some widgets like Scaffold manipulate the MediaQueryData for descendants,
+    // which means the Overlay, which can consider the whole View to 'float'
+    // over the widget tree, may not be aware of obstructions like the software
+    // keyboard. This provides the accurate MediaQueryData for the View to the
+    // Overlay.
+    return MediaQuery.fromView(view: View.of(context), child: portal);
```

Comment:
We should be getting the MediaQueryData from the Overlay's BuildContext, not the View, since the overlay child will be shown on the Overlay.

OverlayPortal has 2 children. Only the overlay child should see the MediaQueryData from the Overlay, the other child (the regular child) should see the same MediaQueryData as its parent.

Also I think we should only copyWith the padding / viewInset (and whatever is relevant in MediaQueryData). Things like accessibility settings should still be from the parent.

Reply:
Here's what I think..

> Only the overlay child should see the MediaQueryData from the Overlay,

I think this means overlayChildBuilder

> the other child (the regular child) should see the same MediaQueryData as its parent.

I think this means child

Reply:
Yeah `overlayChildBuilder` builds the child that will be shown in the Overlay (as if it was in an OverlayEntry) while `child` is just like any other `Widget.child`